backend/libinput/pointer.c

```diff
@@ -136,3 +136,117 @@ void handle_pointer_axis(struct libinput_event *event,
 		}
 	}
 }
+
+void handle_pointer_swipe_begin(struct libinput_event *event,
+		struct libinput_device *libinput_dev) {
+	struct wlr_input_device *wlr_dev =
+		get_appropriate_device(WLR_INPUT_DEVICE_POINTER, libinput_dev);
+	if (!wlr_dev) {
+		wlr_log(WLR_DEBUG, "Got a pointer gesture event for a device with no pointers?");
+		return;
+	}
+	struct libinput_event_gesture *gevent =
+		libinput_event_get_gesture_event(event);
+	struct wlr_event_pointer_swipe_begin wlr_event = { 0 };
+	wlr_event.device = wlr_dev;
+	wlr_event.time_msec =
+		usec_to_msec(libinput_event_gesture_get_time_usec(gevent));
+	wlr_event.fingers = libinput_event_gesture_get_finger_count(gevent);
+	wlr_signal_emit_safe(&wlr_dev->pointer->events.swipe_begin, &wlr_event);
+}
+
+void handle_pointer_swipe_update(struct libinput_event *event,
+		struct libinput_device *libinput_dev) {
+	struct wlr_input_device *wlr_dev =
+		get_appropriate_device(WLR_INPUT_DEVICE_POINTER, libinput_dev);
+	if (!wlr_dev) {
+		wlr_log(WLR_DEBUG, "Got a pointer gesture event for a device with no pointers?");
+		return;
+	}
+	struct libinput_event_gesture *gevent =
+		libinput_event_get_gesture_event(event);
+	struct wlr_event_pointer_swipe_update wlr_event = { 0 };
```

Comment:
You can use struct initializers here.

```c
struct wlr_event_pointer_swipe_update wlr_event = {
	.device = wlr_dev,
	…
};
```

(The rest of the file should be fixed, but this probably belongs to another PR)